From e4579424942fdaf1f9877d36dda2e6b9f1796b03 Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Thu, 18 Apr 2019 15:42:17 +0300
Subject: [PATCH 09/19] Fix updating tablet pressure resolution on every
 proximity enter event

The user can switch pressure sensitivity level in the driver,
which will make our saved values invalid (this option is
provided by Wacom drivers for compatibility reasons, and
it can be adjusted on the fly)

See the bug: https://bugs.kde.org/show_bug.cgi?id=391054

Change-Id: I6cfdff27eaf5a587bf714871f1495a7ea150c553
---
 src/plugins/platforms/windows/qwindowstabletsupport.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/plugins/platforms/windows/qwindowstabletsupport.cpp b/src/plugins/platforms/windows/qwindowstabletsupport.cpp
index 02455536fe..e9d10a2b03 100644
--- a/src/plugins/platforms/windows/qwindowstabletsupport.cpp
+++ b/src/plugins/platforms/windows/qwindowstabletsupport.cpp
@@ -519,6 +519,14 @@ bool QWindowsTabletSupport::translateTabletProximityEvent(WPARAM /* wParam */, L
     if (m_currentDevice < 0) {
         m_currentDevice = m_devices.size();
         m_devices.push_back(tabletInit(uniqueId, cursorType));
+    } else {
+        /**
+         * The user can switch pressure sensitivity level in the driver,
+         * which will make our saved values invalid (this option is
+         * provided by Wacom drivers for compatibility reasons, and
+         * it can be adjusted on the fly)
+         */
+        m_devices[m_currentDevice] = tabletInit(uniqueId, cursorType);
     }
 
     /**
-- 
2.20.1.windows.1

